config/config.go

```diff
@@ -639,47 +644,85 @@ func getStakingTLSCert(v *viper.Viper) (tls.Certificate, error) {
+	_, err := os.Stat(signingKeyPath)
+	keyFileNotFound := errors.Is(err, fs.ErrNotExist)
+
+	switch {
+
+	case ephemeralSignerEnabled && !contentKeyExists && !keyPathIsSet && !rpcSignerURLExists:
+		signer, err := localsigner.New()
 		if err != nil {
-			return nil, fmt.Errorf("couldn't generate ephemeral signing key: %w", err)
+			return nil, fmt.Errorf("couldn't generate ephemeral signing signer: %w", err)
 		}
-		return key, nil
-	}
 
-	if v.IsSet(StakingSignerKeyContentKey) {
+		return signer, nil
+
+	case !ephemeralSignerEnabled && contentKeyExists && !keyPathIsSet && !rpcSignerURLExists:
 		signerKeyRawContent := v.GetString(StakingSignerKeyContentKey)
 		signerKeyContent, err := base64.StdEncoding.DecodeString(signerKeyRawContent)
+
 		if err != nil {
 			return nil, fmt.Errorf("unable to decode base64 content: %w", err)
 		}
-		key, err := localsigner.FromBytes(signerKeyContent)
+
+		signer, err := localsigner.FromBytes(signerKeyContent)
 		if err != nil {
 			return nil, fmt.Errorf("couldn't parse signing key: %w", err)
 		}
-		return key, nil
-	}
 
-	signingKeyPath := getExpandedArg(v, StakingSignerKeyPathKey)
-	_, err := os.Stat(signingKeyPath)
-	if !errors.Is(err, fs.ErrNotExist) {
+		return signer, nil
+
+	case !ephemeralSignerEnabled && !contentKeyExists && keyPathIsSet && !rpcSignerURLExists:
+		// If the key is set, but a user-file isn't provided, we don't create one.
+		// The siging key is only stored to the default file-location if it's created
+		// and saved by the current application run.
+		if keyFileNotFound {
+			return nil, errMissingStakingSigningKeyFile
+		}
+
 		signingKeyBytes, err := os.ReadFile(signingKeyPath)
 		if err != nil {
 			return nil, err
 		}
-		key, err := localsigner.FromBytes(signingKeyBytes)
+		signer, err := localsigner.FromBytes(signingKeyBytes)
 		if err != nil {
 			return nil, fmt.Errorf("couldn't parse signing key: %w", err)
 		}
-		return key, nil
-	}
 
-	if v.IsSet(StakingSignerKeyPathKey) {
-		return nil, errMissingStakingSigningKeyFile
+		return signer, nil
+
+	case !ephemeralSignerEnabled && !contentKeyExists && !keyPathIsSet && rpcSignerURLExists:
+		rpcSignerURL := v.GetString(StakingRPCSignerKey)
+
+		// the rpc-signer client should call a proxy server (on the same machine) that forwards
+		// the request to the actual signer instead of relying on tls-credentials
+		conn, err := grpc.NewClient(rpcSignerURL, grpc.WithTransportCredentials(insecure.NewCredentials()))
+		if err != nil {
+			return nil, fmt.Errorf("couldn't create rpc signer client: %w", err)
+		}
+
+		signer, err := rpcsigner.NewClient(ctx, conn)
+		if err != nil {
+			conn.Close()
+			return nil, fmt.Errorf("couldn't create rpc signer client: %w", err)
+		}
+
+		return signer, nil
+
+	case ephemeralSignerEnabled || contentKeyExists || keyPathIsSet || rpcSignerURLExists:
+		return nil, errInvalidSignerConfig
 	}
 
-	key, err := localsigner.New()
+	// no config values explicitly set,
```

Comment:
This is the existing code, but I think we should just fix it here. If the usage of the ephemeral signer is the default, should this be the default case in the switch + merged w/ the ephemeral signer flag is specified case?